compiler/src/main/java/org/qbicc/graph/literal/LiteralFactory.java

```diff
@@ -240,6 +242,13 @@ public Literal valueConvertLiteral(final Literal value, final WordType toType) {
                 Assert.checkNotNullParam("toType", toType);
                 return value.convert(this, toType);
             }
+
+            @Override
+            public Literal elementOfLiteral(Literal value, Literal index) {
+                Assert.checkNotNullParam("value", value);
+                Assert.checkNotNullParam("index", index);
+                return value.elementOf(this, index);
```

Comment:
Not really because `value` can just be a pointer type and there is no way to determine if index is valid or not.
Though I think I should add a check for `value` to be of `PointerType` or `ArrayType`.